esphome/components/pca9554/pca9554.cpp

```diff
@@ -26,14 +26,25 @@ void PCA9554Component::setup() {
   this->config_mask_ = 0;
   // Invert mask as the part sees a 1 as an input
   this->write_register_(CONFIG_REG, ~this->config_mask_);
-  // All ouputs low
+  // All outputs low
   this->output_mask_ = 0;
   this->write_register_(OUTPUT_REG, this->output_mask_);
   // Read the inputs
   this->read_inputs_();
   ESP_LOGD(TAG, "Initialization complete. Warning: %d, Error: %d", this->status_has_warning(),
            this->status_has_error());
 }
+
+void PCA9554Component::loop() {
+  // Called approximately 10 ms on average (100 Hz). Note: during power up
+  // initialization it will be called at a higher rate.
+  // The read_inputs_() method will cache the input values from the chip.
+  this->read_inputs_();
+  // Clear all the previously read flags.
+  for (int i = 0; i < 8; i++)
+    this->was_previously_read_[i] = false;
```

Comment:
This should be a little bit better
```suggestion
  std::fill(this->was_previously_read_.begin(), this->was_previously_read_.end(), false);
```

Also add `#include <algorithm>` at the top